CMakeLists.txt

```diff
@@ -821,3 +826,32 @@ IF(CPUINFO_SUPPORTED_PLATFORM AND CPUINFO_BUILD_TOOLS)
     INSTALL(TARGETS cpuid-dump RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
   ENDIF()
 ENDIF()
+
+# ---[ pkg-config manifest. This is mostly from JsonCpp...
+if(CPUINFO_BUILD_PKG_CONFIG AND (CPUINFO_LIBRARY_TYPE STREQUAL "default" OR CPUINFO_LIBRARY_TYPE STREQUAL "shared"))
+
+  function(join_paths joined_path first_path_segment)
+    set(temp_path "${first_path_segment}")
+    foreach(current_segment IN LISTS ARGN)
+      if(NOT ("${current_segment}" STREQUAL ""))
+        if(IS_ABSOLUTE "${current_segment}")
+          set(temp_path "${current_segment}")
+        else()
+          set(temp_path "${temp_path}/${current_segment}")
+        endif()
+      endif()
+    endforeach()
+    set(${joined_path} "${temp_path}" PARENT_SCOPE)
+  endfunction()
+
+join_paths(libdir_for_pc_file "\${exec_prefix}" "${CMAKE_INSTALL_LIBDIR}")
```

Comment:
I'm concerned that this wouldn't work on Windows due to Unix environment variable expansions. If it is not supposed to work on Windows, set `CPUINFO_BUILD_PKG_CONFIG` to `OFF` for this platform.

Reply:
`pkg-config` is supported on Windows and is used with MSVC. But I don't have access to a Windows machine to test. I do know that JsonCpp generates a pkg-config manifest [by default](https://github.com/open-source-parsers/jsoncpp/blob/master/CMakeLists.txt#L88) and it is supported on Windows.